functionary/inference.py

```diff
@@ -58,6 +59,16 @@ def prepare_messages_for_inference(
     final_prompt = prompt_template.get_prompt_from_messages(
         dic_messages, tools_or_functions=tools_or_functions
     )
+
+    if tool_choice is not None and tool_choice != "auto":
+        if tool_choice == "none":
+            final_prompt += prompt_template.get_predefined_function_names()[0]
```

Comment:
So I assume that this feature is only applicable to v2 ? Actually we don't need to implement this for V1

Reply:
Actually do you think if tool_choice=="none" we will also set: tools_or_functions=[] ? and then re-generate final_prompt ?

Reply:
1. "So I assume that this feature is only applicable to v2 ? Actually we don't need to implement this for V1"
Yes, this feature is only applicable to v2 since we are not going to enforce backward-compatibility with V1 anymore.

2. "Actually do you think if tool_choice=="none" we will also set: tools_or_functions=[] ? and then re-generate final_prompt ?"
Actually, this is already done [here](https://github.com/MeetKai/functionary/blob/force-function/server_vllm.py#L249-L250) so an empty list for `tools_or_functions` should be passed in here already. I pass in empty list to make sure no functions are in the system message and also add "all\n" with the newline to make sure the model doesn't extend "all".